
# HG changeset patch
# User Carlos R. Mafra <crmafra@gmail.com>
# Date 1253011934 -7200
# Node ID 9420959defee1a8936a715d6b5b42183eaef2e2c
# Parent d58ab202df60c2429cc1e8577e19710b706d2afa
Fix the call to XChangeProperty() in 64-bit mode

    The man page of XChangeProperty() says:

      "If the specified format is 32, the property data must be a
       long array."

    And as we call it with format 32, the type of 'data' must
    be 'long'. It happens to work nowadays in 32-bit architectures
    because sizeof(CARD32) = sizeof(long), but that is no longer
    true in 64-bit mode.

    This patch was downloaded from
    www.openbsd.org/cgi-bin/cvsweb/ports/x11/windowmaker/patches/patch-WINGs_wwindow_c
    and I thank Alexey I. Frolov and Vladimir Nadvornik for helping me
    to understand it on a wmaker-dev thread.

Transplanted from git://repo.or.cz/wmaker-crm.git
commit c7f2a189c48bd4c9eb87958fff66b52e0fdcb7ce

--- a/WINGs/wwindow.c	Mon Aug 24 10:10:03 2009 +0200
+++ b/WINGs/wwindow.c	Tue Sep 15 12:52:14 2009 +0200
@@ -254,14 +254,14 @@
 setMiniwindow(WMWindow *win, RImage *image)
 {
     WMScreen *scr= win->view->screen;
-    CARD32 *data;
+    long *data;
     int x, y;
     int o;
 
     if (!image)
       return;
 
-    data = wmalloc((image->width * image->height + 2) * sizeof(CARD32));
+    data = wmalloc((image->width * image->height + 2) * sizeof(long));
 
     o= 0;
     data[o++] = image->width;
@@ -269,7 +269,7 @@
 
     for (y= 0; y < image->height; y++) {
         for (x= 0; x < image->width; x++) {
-            CARD32 pixel;
+            long pixel;
             int offs= (x+y*image->width);
             
             if (image->format == RRGBFormat)

